inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/http/DefaultEvent2HttpRequestHandler.java

```diff
@@ -0,0 +1,134 @@
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sort.standalone.sink.http;
+
+import org.apache.inlong.sort.standalone.channel.ProfileEvent;
+import org.apache.inlong.sort.standalone.utils.InlongLoggerFactory;
+import org.apache.inlong.sort.standalone.utils.UnescapeHelper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.slf4j.Logger;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DefaultEvent2HttpRequestHandler implements IEvent2HttpRequestHandler {
+
+    public static final Logger LOG = InlongLoggerFactory.getLogger(DefaultEvent2HttpRequestHandler.class);
+
+    @Override
+    public HttpRequest parse(HttpSinkContext context, ProfileEvent event)
+            throws URISyntaxException, JsonProcessingException {
+        String uid = event.getUid();
+        HttpIdConfig idConfig = context.getIdConfig(uid);
+        if (idConfig == null) {
+            context.addSendResultMetric(event, context.getTaskName(), false, System.currentTimeMillis());
+            return null;
+        }
+        // get the delimiter
+        String delimiter = idConfig.getSeparator();
+        char cDelimiter = delimiter.charAt(0);
+        // for tab separator
+        byte[] bodyBytes = event.getBody();
+        String strContext = new String(bodyBytes, Charset.defaultCharset());
+        // unescape
+        List<String> columnValues = UnescapeHelper.toFiledList(strContext, cDelimiter);
+        int valueLength = columnValues.size();
+        List<String> fieldList = idConfig.getFieldList();
+        int columnLength = fieldList.size();
+        // get field value
+        Map<String, String> fieldMap = new HashMap<>();
+        for (int i = 0; i < columnLength; ++i) {
+            String fieldName = fieldList.get(i);
+            String fieldValue = i < valueLength ? columnValues.get(i) : "";
+            fieldMap.put(fieldName, fieldValue);
+        }
+
+        // build
+        String uriString = context.getBaseUrl() + idConfig.getPath();
+        if (!uriString.startsWith("http://") && !uriString.startsWith("https://")) {
+            uriString = "http://" + uriString;
+        }
+        URI uri = new URI(uriString);
+        String jsonData = new ObjectMapper().writeValueAsString(fieldMap);
```

Comment:
No need to init object mapper for every single data

Reply:
fixed.